cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -1231,75 +1262,106 @@ CacheAllocator<CacheTrait>::findEviction(PoolId pid, ClassId cid) {
   // Keep searching for a candidate until we were able to evict it
   // or until the search limit has been exhausted
   unsigned int searchTries = 0;
-  auto itr = mmContainer.getEvictionIterator();
   while ((config_.evictionSearchTries == 0 ||
```

Comment:
I'm a bit confused here. Are we trying to advance the iterator inside or outside the lambda function passed into `withEvictionIterator`?
My guess would be inside, because the second time you call `withEvictionIterator`, the iterator is a newly created one starting from the end of the queue. But this code seems to attempt to advance iterator both inside and outside that lambda.

Reply:
Right, this is because in case of chained items, we might end up evicting the old parent, and the releaseBackToAllocator on line 1358 will return `kNotRecycled`. In that case, we'll just loop again and start looking for a new candidate (calling `withEvictionIterator` again). I did it this way to keep the behavior as close as possible to the original implementation.